### PR TITLE
Migrate to authentication policy v1alpha2

### PR DESCRIPTION
**Pull request by diemtvu**

### CODE REVIEW

File: pilot/pkg/config/kube/crd/types.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ var knownTypes = map[string]schemaType{
 		schema: model.ServiceRole,
 		object: &ServiceRole{
 			TypeMeta: meta_v1.TypeMeta{
-				Kind:       model.ServiceRole.Type,
+				Kind:       "ServiceRole",
```
Comment (kyessenov) [inline pilot/pkg/config/kube/crd/types.go]: This file is auto-generated, why is it edited?

Comment (diemtvu) [reply pilot/pkg/config/kube/crd/types.go]: I think someone do it manually previously. This is output when I rerun the script. I can remove this from PR as my api change doesn't affect it.

**Comment by diemtvu** [comment]
The PR is complete now. Can I have new lgtm. Thanks.

**Comment by nmittler** [comment]
@diemtvu After this change, I'm getting this error from the [new pilot e2e tests](https://github.com/istio/istio/pull/4296):

```
I'm getting this error when running the new pilot e2e tests: 2018-03-21T21:48:23.944401Z	info Command error: exit status 1
authn_policy_test.go:28: command failed: "Error from server (BadRequest): error when creating \"testdata/v1alpha1/authn-policy.yaml.tmpl\": the API version in the data (authentication.istio.io/v1alpha2) does not match the expected API version (authentication.istio.io/v1alpha1)\nError from server (BadRequest): error when creating \"testdata/v1alpha1/authn-policy.yaml.tmpl\": the API version in the data (authentication.istio.io/v1alpha2) does not match the expected API version (authentication.istio.io/v1alpha1)\n" exit status 1
```

I've rebased onto master .... is there anything else I need to do?

**Comment by kyessenov** [comment]
It's not allowed to have two versions for the API group. Delete your old
CRD definition for v1alpha1.

**Pull request merged**